Bottom: fc9fdf9656a2808a369705c93aa6f76ca46c0dd5
Top:    49260d681270041f97193138fbc7f4c452b83b83
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-13 01:52:03 +0100

Introduce COORD_MIN macro to avoid open-coding the minumum as -COORD_MAX - 1


---

diff --git a/configure.ac b/configure.ac
index 9afffd8..1842764 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1093,18 +1093,22 @@ case "$enable_coord32:$enable_coord64:$ac_cv_header_stdint_h" in
   yes:no:yes )
     COORD_TYPE="int32_t"
     COORD_MAX="INT32_MAX"
+    COORD_MIN="INT32_MIN"
     ;;
   no:yes:yes )
     COORD_TYPE="int64_t"
     COORD_MAX="INT64_MAX"
+    COORD_MIN="INT64_MIN"
     ;;
   yes:no:no  )
     COORD_TYPE="int"
     COORD_MAX="INT_MAX"
+    COORD_MIN="INT_MIN"
     ;;
   no:yes:no  )
     COORD_TYPE="long long"
     COORD_MAX="LLONG_MAX"
+    COORD_MIN="LLONG_MIN"
     ;;
   yes:yes:*  )
     AC_MSG_ERROR("*** cannot require both 32 and 64 bit coordinates")
@@ -1112,12 +1116,15 @@ case "$enable_coord32:$enable_coord64:$ac_cv_header_stdint_h" in
   *:*:*    )
     COORD_TYPE="long"
     COORD_MAX="LONG_MAX"
+    COORD_MIN="LONG_MIN"
     ;;
 esac
 AC_DEFINE_UNQUOTED([COORD_TYPE],[$COORD_TYPE],
   [C type for Coordinates])
 AC_DEFINE_UNQUOTED([COORD_MAX],[$COORD_MAX],
   [Maximum value of coordinate type])
+AC_DEFINE_UNQUOTED([COORD_MIN],[$COORD_MIN],
+  [Minimum value of coordinate type])
 
 # ------------- Complete set of CPPFLAGS and LIBS -------------------
